docarray/document/mixins/multimodal.py

```diff
@@ -41,12 +46,48 @@ def _from_dataclass(cls, obj) -> 'Document':
                     'type': field.type.__name__,
                 }
 
-            elif field.type == bytes and not _is_field(field):
-                tags[key] = base64.b64encode(attribute).decode()
+            elif field.type in [
+                Optional[str],
+                Optional[int],
+                Optional[float],
+                Optional[bool],
+            ] and not _is_field(field):
+                tags[key] = attribute
                 multi_modal_schema[key] = {
                     'attribute_type': AttributeType.PRIMITIVE,
-                    'type': field.type.__name__,
+                    'type': f'Optional[{_get_optional_subtype(field.type).__name__}]',
+                }
+
+            elif field.type in [bytes, Optional[bytes]] and not _is_field(field):
+                tags[key] = (
+                    base64.b64encode(attribute).decode() if attribute else attribute
+                )
+                multi_modal_schema[key] = {
+                    'attribute_type': AttributeType.PRIMITIVE,
+                    'type': field.type.__name__
+                    if not (_is_optional(field.type))
+                    else f'Optional[{_get_optional_subtype(field.type).__name__}]',
```

Comment:
Here I am open to suggestion should I just remove the Optional in the type name ?